src/Swarm/Log.hs

```diff
@@ -39,35 +30,44 @@ import Swarm.Game.Universe (Cosmic)
 
 -- | Severity of the error - critical errors are bugs
 --   and should be reported as Issues.
-data ErrorLevel = Debug | Warning | Error | Critical
+data Severity = Info | Debug | Warning | Error | Critical
   deriving (Show, Eq, Ord, Generic, FromJSON, ToJSON)
 
--- | Source of the robot log.
-data LogSource
-  = -- | Log produced by 'Swarm.Language.Syntax.Say'
+-- | How a robot log entry was produced.
+data RobotLogSource
+  = -- | Produced by 'Swarm.Language.Syntax.Say'
     Said
-  | -- | Log produced by 'Swarm.Language.Syntax.Log'
+  | -- | Produced by 'Swarm.Language.Syntax.Log'
     Logged
+  | -- | Produced as the result of an error.
+    RobotError
+  deriving (Show, Eq, Ord, Generic, FromJSON, ToJSON)
+
+-- | Source of a log entry.
+data LogSource
+  = -- | Log produced by a robot.  Stores information about which
+    --   command was used and the ID of the producing robot.
+    RobotLog RobotLogSource Int
   | -- | Log produced by an exception or system.
-    ErrorTrace ErrorLevel
+    SystemLog
   deriving (Show, Eq, Ord, Generic, FromJSON, ToJSON)
 
 data LogLocation a = Omnipresent | Located a
   deriving (Show, Eq, Ord, Generic, FromJSON, ToJSON)
 
--- | An entry in a robot's log.
+-- | A log entry.
 data LogEntry = LogEntry
   { _leTime :: TickNumber
   -- ^ The time at which the entry was created.
   --   Note that this is the first field we sort on.
   , _leSource :: LogSource
-  -- ^ Whether this log records a said message.
-  , _leRobotName :: Text
-  -- ^ The name of the robot that generated the entry.
-  , _leRobotID :: Int
-  -- ^ The ID of the robot that generated the entry.
+  -- ^ Where this log message came from.
+  , _leSeverity :: Severity
+  -- ^ Severity level of this log message.
+  , _leName :: Text
+  -- ^ Name of the robot or subsystem that generated this log entry.
   , _leLocation :: LogLocation (Cosmic Location)
-  -- ^ Location of the robot at log entry creation.
+  -- ^ Location associated with this log message.
```

Comment:
Maybe we should move location to robot log source. 🤔

Reply:
@byorgey I would say those are different things - system message can talk _about_ a location that is not its own.

If we need to show message only to certain location, using a robot seems like the natural (and in scenarios easier) way to do it.

Either way is fine, but I suspect it will feel weird having to set default location each time a system logs something.

Reply:
OK, I buy that.  I've updated it to put the location exclusively in robot log messages.